clippy_lints/src/print.rs

```diff
@@ -78,12 +78,28 @@ declare_lint! {
     "use of `Debug`-based formatting"
 }
 
+/// **What it does:** This lint warns about the use of literals as `print!`/`println!` args.
+///
+/// **Why is this bad?** Using literals as `println!` args is inefficient
+/// (c.f., https://github.com/matthiaskrgr/rust-str-bench) and unnecessary
+/// (i.e., just put the literal in the format string)
+///
+/// **Example:**
+/// ```rust
+/// println!("{}", "foo");
+/// ```
+declare_lint! {
```

Comment:
We're deprecating the use of this macro. Please use `declare_clippy_lint` together with the nursery group

Reply:
This is fixed in 511aa65.